### PR TITLE
add TX/RX packet dump option

### PR DESCRIPTION
**Pull request by mszczodrak**

### CODE REVIEW

File: src/core/mac/mac.cpp [hunk 1]
```diff
@@ -1343,6 +1344,7 @@ void Mac::ReceiveDoneTask(Frame *aFrame, ThreadError aError)
         {
             for (Receiver *receiver = mReceiveHead; receiver; receiver = receiver->mNext)
             {
+                otDumpDebgMac("RX", aFrame->GetHeader(), aFrame->GetLength());
```
Comment (jwhui) [inline src/core/mac/mac.cpp]: Do we need to print the frame buffer for each receiver?

File: src/core/mac/mac.cpp [hunk 2]
```diff
@@ -1018,6 +1018,7 @@ void Mac::SentFrame(ThreadError aError)
         sender->mNext = NULL;
 
         mDataSequence++;
+        otDumpDebgMac("TX", sendFrame.GetHeader(), sendFrame.GetLength());
```
Comment (jwhui) [inline src/core/mac/mac.cpp]: Should we print all transmitted frame or just the data frames?

File: src/core/mac/mac.cpp (hunk 1 above)
Comment (mszczodrak) [inline src/core/mac/mac.cpp]: I would suggest to add another packet dump option strictly on the data path, thus allowing folks to either poke across all packets (here) or just poke on the data path.

Comment (jwhui) [inline src/core/mac/mac.cpp]: In this case, I was wondering if the dump should be outside the for-loop.  Inside is useful if the handlers modify the frame buffer.

Comment (mszczodrak) [inline src/core/mac/mac.cpp]: Good idea.

**Pull request merged**